v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,943 @@
+   *
+   * @param colName - The column name used to fetch the key from {@code valuesJson}.
+   * @param valuesJson - The {@link JSONObject} containing all the key-value pairs for the current
+   *     incoming stream.
+   * @return A {@link BigInteger} object representing the varint value from the Cassandra data.
+   * @throws IllegalArgumentException If the value is not a valid format for varint (neither a valid
+   *     number string nor a byte array).
+   */
+  public static BigInteger handleCassandraVarintType(String colName, JSONObject valuesJson) {
+    Object value = valuesJson.get(colName);
+
+    if (value instanceof String) {
+      try {
+        return new BigInteger((String) value);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException(
+            "Invalid varint format (string) for column: " + colName, e);
+      }
+    } else if (value instanceof byte[]) {
+      try {
+        return new BigInteger((byte[]) value);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(
+            "Invalid varint format (byte array) for column: " + colName, e);
+      }
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Generates a {@link Duration} based on the provided {@link CassandraTypeHandler}.
+   *
+   * <p>This method fetches a string value from the provided {@code valuesJson} object using the
+   * column name {@code colName}, and converts it into a {@link Duration} object. The string value
+   * should be in the ISO-8601 duration format (e.g., "PT20.345S").
+   *
+   * @param colName - The column name used to fetch the key from {@code valuesJson}.
+   * @param valuesJson - The {@link JSONObject} containing all the key-value pairs for the current
+   *     incoming stream.
+   * @return A {@link Duration} object representing the duration value from the Cassandra data.
+   * @throws IllegalArgumentException if the value is not a valid duration string.
+   */
+  public static Duration handleCassandraDurationType(String colName, JSONObject valuesJson) {
+    String durationString = valuesJson.optString(colName, null);
+    if (durationString == null) {
+      return null;
+    }
+    try {
+      return Duration.parse(durationString);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Invalid duration format for column: " + colName, e);
+    }
+  }
+
+  /**
+   * Generates a Type based on the provided {@link CassandraTypeHandler}.
+   *
+   * @param colName - which is used to fetch Key from valueJSON.
+   * @param valuesJson - contains all the key value for current incoming stream.
+   * @return a {@link InetAddress} object containing InetAddress as value represented in cassandra
+   *     type.
+   */
+  public static InetAddress handleCassandraInetAddressType(String colName, JSONObject valuesJson)
+      throws UnknownHostException {
+    return InetAddress.getByName(valuesJson.getString(colName));
```

Comment:
Please consider using [com.google.common.net.InetAddresses#forString](https://guava.dev/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-] instead of `InetAddress.getByName`

Motivation:
The [InetAddress.html#getByName(java.lang.String)](https://docs.oracle.com/javase/7/docs/api/java/net/InetAddress.html#getByName(java.lang.String)) That's used here tries to resolve a host name into it's IP address via a DNS query. 

Most likely customer's Dataflow workers will not have firewall permissions to do do that.  `getByName` will cause the code to throw [UnknownHostException](https://docs.oracle.com/javase/7/docs/api/java/net/UnknownHostException.html) which we are not catching here.
Depending on the configuration of the security manager on customer's jvm, `getByName` can also throw  [SecurityException](https://docs.oracle.com/javase/7/docs/api/java/lang/SecurityException.html) both of which we would like to avoid having to deal with.

The `google.common` equivalent [com.google.common.net.InetAddresses#forString](https://guava.dev/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-]  does not do any DNS query and simply throws [IllegalArgumentException](http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true) if the string isn't a valid ipaddress literal.

Reply:
Thanks. This has been address now.